From f755a2214f3453609ca3e1b0593314ee0a2fe6f4 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Sat, 29 Jan 2022 16:21:26 +0200
Subject: [PATCH] [SW-65638] gaudi2: release: adjust the operation calls

After lining up all the *_op functions to check the flag existence
and then check and operate the *_args. The the caller layer can
unified the way it calls those commands

Change-Id: I5d4041131ff6247aedf65d3fc091aab28bb6f4e1
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/hls_platform.rc b/scripts/fwi/hls_platform.rc
index ca80e5c..a347904 100644
--- a/scripts/fwi/hls_platform.rc
+++ b/scripts/fwi/hls_platform.rc
@@ -1080,45 +1080,27 @@
 function sys_special_commands()
 {
 #	pinfo -l 3  "sys_special_commands"
-	[ ${show_status_f:-0} -eq 1 ] && status_op
-
 	[ ${show_pci_map_f:-0} -eq 1 ] &&  hlp_redirect hls2_oam2pci_show
-	[ "$i2c_scan" != "" ] && {
-		oam_i2c_scan_op
-		[ $? -ne 0 ] && return 1
-	}
 
-	[ "$i2c_send" != "" ] && {
-		bmc_i2c_send_op
-		[ $? -ne 0 ] && return 1
-	}
-
-	[ "$bmc_raw" != "" ] && {
-		bmc_raw_op
-		[ $? -ne 0 ] && return 1
-	}
-
-	[ "$set_pci_sw_f" != "" ] && {
-		set_pci_sw
-		[ $? -ne 0 ] && {
-			perr "Failed to set PCIe switches"
-		}
-	}
-	# disable/enable WP devices
-	oams_wp_op
-	oams_set_recovery_mode_op
-	oams_set_btl_mode_op
-	oams_set_uart_op
-	oams_full_pwr_op
-	oams_rst_op
+	status_op || retrun 1
+	oam_i2c_scan_op || return 1
+	bmc_i2c_send_op || return 1
+	bmc_raw_op || return 1
+	set_pci_sw_op || return 1
+	oams_wp_op || return 1
+	oams_set_recovery_mode_op || return 1
+	oams_set_btl_mode_op || return 1
+	oams_set_uart_op || return 1
+	oams_full_pwr_op || return 1
+	oams_rst_op || return 1
 
 	[ ${host_reboot_f:-0} -eq 1 ] && {
 		pinfo "[HOST] start reboot"
 		host_run_sudo reboot
 	}
 
-	sys_pwr_op
-	his_pwr_op
+	sys_pwr_op || return 1
+	his_pwr_op || return 1
 
 	return 0
 }
